base/threadingconstructs.jl

```diff
@@ -344,9 +351,9 @@ macro spawn(args...)
         let $(letargs...)
             local task = Task($thunk)
             task.sticky = false
-            local tpid_actual = $tpid
+            local tpid_actual = _sym_to_tpid($(QuoteNode(tp)))
             if _nthreads_in_pool(tpid_actual) == 0
-                tpid_actual = Int8(0)
+                tpid_actual = _sym_to_tpid(:default)
             end
             ccall(:jl_set_task_threadpoolid, Cint, (Any, Int8), task, tpid_actual)
```

Comment:
Can we encapsulate all this in a function instead of emitting a new copy in each macro use?